src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java

```diff
@@ -831,6 +831,41 @@ public void testOutOfRangesArguments(final String[] argv) throws Exception {
         clp.parseArguments(System.err, argv);
     }
 
+    @Test(expectedExceptions = CommandLineException.CommandLineParserInternalException.class)
+    public void testHiddenRequiredArgumentThrowException() throws Exception {
+        @CommandLineProgramProperties(summary = "tool with required and hidden argument",
+                oneLineSummary = "tool with required and hidden argument",
+                programGroup = TestProgramGroup.class)
+        final class ToolWithRequiredHiddenArgument {
+            // TODO: this does not blow up if it is a primitive, because it is not consider required
```

Comment:
Ok, I overseen that primitives are never null and this is point out in the javadoc. I think that this could be fixed (in a different PR) by checking if the field has a primitive type to determine if it is optional or not. I removed this comment and I will open an issue.